sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -921,25 +930,20 @@ private long insertRevision(SQLDatabase db,InsertRevisionOptions options) {
             logger.fine("New revision inserted: " + options.docNumericId + ", " + options.revId);
             newSequence = db.insert("revs", args);
             if (newSequence < 0) {
-                throw new IllegalStateException("Unknown error inserting new updated doc, please checking log");
+                throw new IllegalStateException("Unknown error inserting new updated doc, please checking log"); //TODO is this recoverable?
```

Comment:
newSequence < 0 means the row couldn't be inserted, which is probably something quite serious and probably not recoverable. Also take the opportunity to clean up the grammar on this error message.